Automattic/newspack-blocks
# fix: disambiguate users and guest authors in Author List exclusions

### All Submissions:






### Changes proposed in this Pull Request:



#1143 fixed a bug in the Author Profile block when authors and guest authors had the same ID. This PR fixes a similar bug in the exclusions attribute of the Author List block. However, it also requires https://github.com/Automattic/newspack-plugin/pull/1679 to be released to the `newspack-components` NPM package, so consider this a draft PR until that release is deployed.

Note that this fix won't fix pre-existing instances of the block, although it does handle the flat array structure of the attribute for backwards compatibility. Any blocks affected by the bug will need to be rebuilt.

Closes #1150.

### How to test the changes in this Pull Request:

1. In a local environment, checkout https://github.com/Automattic/newspack-plugin/pull/1679 and this PR.
2. Confirm that #1150 is not replicable: only the exact user or guest author you select should be checked in the block controls, and only that user or guest author should be excluded in the editor preview and on the front-end.

### Other information:
